### PR TITLE
order creation -  accessibility -  Set headings to order list screen and order creation customer details

### LINKED ISSUE #5815: [Order Creation] Accessibility → Add Customer → Details, Billing Address, etc. are not marked as headings
The section titles like Details and Billing Address are not dictated by TalkBack as headings. This means that it will not be possible to navigate using the Headings reader control.

### PR DESCRIPTION
**Pull request by rossanafmenezes**
Closes: #5815 

### Description

There are two screens involved in this PR: The Order list screen and the Add Customer Details under Order Creation. 
This sets the headings of both screens as accessibility-ready so they can be announced and treated as headings by the TalkBack option. 

A few observations regarding the code:

Initially, the headings were set using the `ViewCompat.setAccessibilityHeading` only, which worked, however, when using the explore by touch option (dragging the finger throughout the screen) the headings can only be read if you go directly to the textview, which is very small. The `announceForAccessibility`  option reads the row (the texview container) which makes it easier to  use the explore by touch option.

### Testing instructions

**Screen 1 - Order list:**

- Enable talkback from the accessibility settings on your device
- Go to the orders tab
There are 3 different gestures to be tested here:

1. tap on each of the headings and confirm it's read "heading text - Heading" (ex. Yesterday - Heading)
2. Try the explore by touch option (tap on any part of the screen and slowly drag one finger around the screen) and make sure the headings can be accessed this way
3. Change the reading control (Talkback version 9.1 and up swipe up using 3 fingers) until you access the headings option. Then swipe up and down using one finger and make sure it reads only (and all) headings without moving to other elements of the screen. 

**Screen 2 - Add Customer Details:**

- Enable talkback from the accessibility settings on your device
- enable order creation under Settings ->  Beta features
- Go to the orders tab
- tap on the + sign to add a new order
- tap on create order
- tap on + Add customer details

Same as before, there are 3 different gestures to be tested here:

1. tap on each of the headings and confirm it's read "heading text - Heading" (ex. Details - Heading)
2. Try the explore by touch option (tap on any part of the screen and slowly drag one finger around the screen) and make sure the headings can be accessed this way
3. Change the reading control (Talkback version 9.1 and up swipe up using 3 fingers) until you access the headings option. Then swipe up and down using one finger and make sure it reads only (and all) headings without moving to other elements of the screen. 

### Images/gif

**Before**

https://user-images.githubusercontent.com/30724184/158692175-7446e854-237e-4405-8fe8-d6f0fa478514.mp4

**After**

https://user-images.githubusercontent.com/30724184/158692718-f63fef47-c03a-4c6a-945a-4614c5e8dc06.mp4

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by wzieba** [comment]
Sorry @rossanafmenezes , this PR slipped out of my radar. I'll review it ~today~ tomorrow.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationCustomerAddFragment.kt [hunk 1]
```diff
@@ -116,9 +116,19 @@ class OrderCreationCustomerAddFragment : BaseFragment(R.layout.fragment_creation
         }
     }
 
+    /*
+   Set the announceForAccessibility function to the heading containers to allow the use of the
+   explore by touch option.
+   */
```
Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationCustomerAddFragment.kt]: I would opt to remove this comment as it kinda indicates that `inflateLayout` is responsible for only for accessibility options.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationCustomerAddFragment.kt [hunk 2]
```diff
@@ -116,9 +116,19 @@ class OrderCreationCustomerAddFragment : BaseFragment(R.layout.fragment_creation
...
     private fun inflateLayout(view: View) {
         billingBinding = LayoutAddressFormBinding.inflate(layoutInflater).apply {
-            addressSectionHeader.setText(R.string.order_detail_billing_address_section)
+            detailsHeaderContainer.announceForAccessibility(R.string.details.toString())
+            ViewCompat.setAccessibilityHeading(detailsHeaderContainer, true)
+
+            addressHeaderContainer.announceForAccessibility(R.string.order_detail_billing_address_section.toString())
+            addressSectionHeader.setText(R.string.order_detail_add_billing_address)
+            ViewCompat.setAccessibilityHeading(addressHeaderContainer, true)
```
Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationCustomerAddFragment.kt]: ❓ Can we maybe extract this to some static method with parameter? I see that in lines 141 the code is roughly the same.

Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationCustomerAddFragment.kt:130]: ❓ Can we maybe extract this to some static method with parameter? I see that in lines 141 the code is roughly the same.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListAdapter.kt [hunk 3]
```diff
@@ -187,9 +187,19 @@ class OrderListAdapter(
...
     private class SectionHeaderViewHolder(val viewBinding: OrderListHeaderBinding) :
-        RecyclerView.ViewHolder(viewBinding.getRoot()) {
+        RecyclerView.ViewHolder(viewBinding.root) {
         fun onBind(header: SectionHeader) {
             viewBinding.orderListHeader.setText(TimeGroup.valueOf(header.title.name).labelRes)
+
+            /*
+            Set the announceForAccessibility function to the heading container to allow the use of the
+            explore by touch option.
+            */
```
Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListAdapter.kt]: ⛏️ I would suggest removing comment as the code below is I believe self-explanatory. But it's a minor!

Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListAdapter.kt:197]: ⛏️ I would suggest removing comment as the code below is I believe self-explanatory. But it's a minor!

**Review by wzieba — COMMENTED** [review]
👋 @rossanafmenezes improvements for list of orders and billing address work fine, but it seems it doesn't work for shipping address:

https://user-images.githubusercontent.com/5845095/159582875-6bbb45b9-2f6a-443e-99c9-48fd9e575a6d.mov

---

second thing is that there's a failing build and I also couldn't build project for the same reason. I think it's the same as the one here (internal link: p1646845001099659-slack-C6H8C3G23). I'll allow myself to merge newest `trunk` to resolve it.

**Review by wzieba — APPROVED** [review]
Sorry @rossanafmenezes ! After your ping I did tests again and I must've confused `Headings` with `Controls` options in Talkback.

So it looks good to me - I'm leaving code-related comments though but they're not blocking.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationCustomerAddFragment.kt (hunk 2 above)
Comment (rossanafmenezes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationCustomerAddFragment.kt]: Done :D

**Pull request merged**